### ISSUE TITLE
javax.servlet dependency not compatible with tomcat 10

### ISSUE THREAD

**Issue by siropa** [leonardehrenfried/jspc-maven-plugin#114]
The pom includes a dependency for javax.servlet-api version 4.0.1.  When compiling for target of tomcat 10, getting errors about missing javax.servlet.jsp.tagext.BodyTagSupport. It should be using jakarta.servlet.jsp.tagext.BodyTagSupport which is in jakarta.servlet-api version 6.0.0.  Can this old dependency be updated please?

**Comment by leonardehrenfried** [comment]
Sure, could you please send a pull request?

**Comment by leonardehrenfried** [comment]
I just released version 4.1.0 which upgrades a bunch of dependencies including the one you mention. Can you check if it solves your problem?

**Comment by leonardehrenfried** [comment]
To be honest, I have no idea how this should work correctly since I haven't touched JSP in over 10 years. If you could send a PR with the correct pom.xml I'm happy to merge it in.

The thing is, there isn't any code in this repo that is referencing `javax/servlet/jsp/tagext/BodyTagSupport`.

Do you maybe write some custom tags like these https://github.com/leonardehrenfried/jspc-maven-plugin/blob/master/src/it/project_multimodule_with%20space/moduleB/src/main/java/com/example/plugin/jspc/tag/HelloTag.java and have an import on the `javax` namespace there?

**Comment by siropa** [comment]
We do have custom tags like that, but they are all using jakarta references.  I even cleaned out my maven repo and then searched everything that was imported with a fresh build for old javax references and didn't find any. I have no idea where that is coming from.

If I had a fix, i'd gladly submit it, but i'm stumped at this point.  When i've got some time i'll keep looking it to. thanks for the help.

**Comment by leonardehrenfried** [comment]
Ah, I remembered something last night. I had to change some namespace definition when moving from javax to jakarta in the test projects.

https://github.com/leonardehrenfried/jspc-maven-plugin/commit/c25931fae083af6a1acde4ec1d858d10909f848c

Do you have old URIs in your JSPs somewhere?

**Comment by siropa** [comment]
Good idea, but no.  to test, I created a fake javax.servlet.jsp.tagext.BodyTagSupport.java file in my project. Ran the jsp compiler and found the issue. We use this old pdf library and apparently that has some imbedded references to the old version.  So my fix is dealing with that 12year old out of data lib. 

thanks for your help and getting the project updated with the newer versions.

**Comment by leonardehrenfried** [comment]
Good detective work!
